### PR TITLE
Improve and clean-up Fio code

### PR DESCRIPTION
**Pull request by rsevilla87**
Improve Fio code with more log messages, better exception handling and code optimizations.

### CODE REVIEW

**Comment by dry923** [comment]
@rsevilla87 rebase please

File: run_snafu.py [hunk 1]
```diff
@@ -38,12 +38,12 @@
...
 def main():
     # collect arguments
-    parser = argparse.ArgumentParser(description="run script")
+    parser = argparse.ArgumentParser(description="run script", add_help=False)
     parser.add_argument(
         '-v', '--verbose', action='store_const', dest='loglevel', const=logging.DEBUG,
         default=logging.INFO, help='enables verbose wrapper debugging info')
     parser.add_argument(
-        '-t', '--tool', help='Provide tool name')
+        '-t', '--tool', help='Provide tool name', required=True)
```
Comment (bengland2) [inline run_snafu.py:46]: I like this, catch mistakes as early as possible.

**Comment by bengland2** [comment]
I'm running this PR on my laptop using minikube and still getting the same error from fio, no indication of why it happens:
```
2020-04-21T19:54:33Z - ERROR    - MainProcess - trigger_fio: Fio failed to execute, trying one more time..
2020-04-21T19:54:34Z - ERROR    - MainProcess - trigger_fio: Fio failed to execute a second time, stopping...
2020-04-21T19:54:34Z - ERROR    - MainProcess - trigger_fio: (b'', 1)
```
where does this last message even come from?

**Comment by bengland2** [comment]
never mind that last comment, I was using the wrong snafu, I used yours and now I get a clear idea of what was wrong:
```
2020-04-22T17:24:20Z - ERROR    - MainProcess - trigger_fio: Fio failed to execute
2020-04-22T17:24:20Z - ERROR    - MainProcess - trigger_fio: Output file: <fio-server-1-benchmark-d0ea5d9f-dk7
qt> fio: output file open error: No such file or directory
<fio-server-2-benchmark-d0ea5d9f-6hvmj> fio: output file open error: No such file or directory
```

**Comment by aakarshg** [comment]
can you please rebase @rsevilla87

**Pull request merged**